_posts/2022-11-07-changelog-182.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #182"
+date: 2022-11-07 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#9650] Increase the minimal supported version of Rust to 1.56
+
+* Procedural macro support improvements:
+  * [#9628] Enable the expansion of [function-like procedural macros] in the [nightly builds].
+    This feature can be controlled via `org.rust.macros.proc.function-like` option in `Experimental Features` dialog.
+    For more information, check the [implementation PR](https://github.com/intellij-rust/intellij-rust/pull/6564) and the corresponding [tracking issue](https://github.com/intellij-rust/intellij-rust/issues/6908).
+
+  * [#9526] Insert brackets of the proper form (`()`, `[]` or `{}`) when completing procedural macro calls
+
+  * [#9614] Fix the expansion of [attribute] and [derive] procedural macros inside functions
+
+  * [#9559] Fix duplicate completion for already imported [attribute] procedural macros
+
+* [#9105] Allow to run a specific [documentation test](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) [all the ways](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-testing.html) you can run a usual test (by [@Kobzol])
+TODO: gif
+
+* [#9349] Decorate MSVC type names when rendering values in the debugger on Windows. Now the Rust type names containing MSVC-specific wrappers are rendered properly, e.g. `tuple$<Foo, ref$<Bar>>` is replaced by `(Foo, &Bar)`.
+  This feature can be disabled via the `Decorate MSVC type names` checkbox in the `File | Settings | Build, Execution, Deployment | Debugger | Data Views | Rust` settings
+TODO: image
+
+* [#7134] Add a quick-fix to generate missing trait member definitions in trait implementations (by [@Sejsel])
+TODO: gif
+
+* [#8755] Add basic parsing support for `let_chains` (by [@Kobzol])
+
+* [#8916] Fix emulating terminal in output console in some cases
+
+## Fixes
+
+* Type inference fixes:
+  * [#9598], [#9639] Fix false-positive `trait bound is not satisfied E0277` annotations.
+    Infer types properly when there are trait bounds on the `Self` type
+   
+  * [#9552] Properly infer type of custom macros that shadow built-in macros
+
+  * [#7450] Properly infer array size when it refers to a constant (by [@Kobzol])
+
+  * [#9586] Infer const arguments in struct literal paths
+
+  * [#9620] Normalize associated type in `impl` item trait ref
+
+  * [#9631] Fix type inference in the case of normalizable associated type projections like `<A as B>::C`
+
+* [#9637] Do not cancel the [build script] evaluation when the native helper binary was not found
```

Comment:
`cancel` is not suitable here, from my point of view. Previously, it just failed